src/watchman/models.clj

```diff
@@ -206,12 +218,38 @@
     (k/set-fields fields)
     (k/where {:id id})))
 
+(defn snooze-msg [snooze-until]
+  "Given a Joda DateTime, return a formatted message for the UI indicating until when this role is snoozed.
+  Formatted string will look like: 'Snoozed until 04-21-15 23:38 PST8PDT'.  Currently PST8PDT is hardcoded in
+  for the timezone."
+  (let [tz (time-core/time-zone-for-id "PST8PDT")
+        formatter (time-format/with-zone (time-format/formatter "MM-dd-yy HH:mm") tz)
+        formatted-time (->> snooze-until
+                            time-coerce/to-date-time
+                            (time-format/unparse formatter))]
+      (str "Snoozed until " formatted-time " " (.getID tz))))
+
+(defn role-snoozed?
+  "Determine if the specified role is still asleep as specified by a snooze. cur-time is provided as an
+  argument so that this result can be made consistent with other timing related queries"
+  ([role] (role-snoozed? role (time-core/now)))
+  ([role cur-time] (let [snooze-until (-?> (:snooze_until role)
+                                           time-coerce/to-date-time)]
+                     (and snooze-until
+                          (time-core/before? cur-time snooze-until)))))
+
 (defn ready-to-perform?
-  "True if enough time has elapsed since we last checked this alert."
+  "True if enough time has elapsed since we last checked this alert and the alert's role isn't
+  snoozed"
   [check-status]
-  (let [last-checked-at (-?> (sget check-status :last_checked_at)
+  (let [cur-time (time-core/now)
+        role-id (->> check-status :checks :role_id)
+        role-snoozed (role-snoozed? (get-role-by-id role-id) cur-time)
+        last-checked-at (-?> (sget check-status :last_checked_at)
                              time-coerce/to-date-time)]
-    (or (nil? last-checked-at)
-        (time-core/after? (time-core/now)
-                          (time-core/plus last-checked-at
-                                          (time-core/secs (sget-in check-status [:checks :interval])))))))
+    (and (not role-snoozed)
```

Comment:
In general prefer `when` to this usage of `and`. Also the `role-snoozed` variable isn't adding much; it would be more concise and readable to just inline the `(role-snoozed? ...)` call.